spring-integration-mail/src/main/java/org/springframework/integration/mail/dsl/ImapIdleChannelAdapterSpec.java

```diff
@@ -0,0 +1,373 @@
+	 */
+	public ImapIdleChannelAdapterSpec headerMapper(HeaderMapper<MimeMessage> headerMapper) {
+		assertReceiver();
+		this.receiver.setHeaderMapper(headerMapper);
+		return _this();
+	}
+
+	/**
+	 * When a header mapper is provided determine whether an embedded {@link Part} (e.g
+	 * {@link Message} or {@link javax.mail.Multipart} content is rendered as a byte[] in
+	 * the payload. Otherwise, leave as a {@link Part}. These objects are not suitable for
+	 * downstream serialization. Default: true.
+	 * <p>
+	 * This has no effect if there is no header mapper, in that case the payload is the
+	 * {@link MimeMessage}.
+	 * @param embeddedPartsAsBytes the embeddedPartsAsBytes to set.
+	 * @return the spec.
+	 * @see #headerMapper(HeaderMapper)
+	 */
+	public ImapIdleChannelAdapterSpec embeddedPartsAsBytes(boolean embeddedPartsAsBytes) {
+		assertReceiver();
+		this.receiver.setEmbeddedPartsAsBytes(embeddedPartsAsBytes);
+		return _this();
+	}
+
+
+	/**
+	 * Configure a {@link TransactionSynchronizationFactory}. Usually used to synchronize
+	 * message deletion with some external transaction manager.
+	 * @param transactionSynchronizationFactory the transactionSynchronizationFactory.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec transactionSynchronizationFactory(
+			TransactionSynchronizationFactory transactionSynchronizationFactory) {
+		this.target.setTransactionSynchronizationFactory(transactionSynchronizationFactory);
+		return this;
+	}
+
+	/**
+	 * Configure a chain of {@link Advice} objects for message delivery.
+	 * @param adviceChain the advice chain.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec adviceChain(Advice... adviceChain) {
+		this.adviceChain.addAll(Arrays.asList(adviceChain));
+		return this;
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} with the provided
+	 * {@code PlatformTransactionManager} and default {@link DefaultTransactionAttribute}
+	 * for the {@code pollingTask}.
+	 * @param transactionManager the {@link PlatformTransactionManager} to use.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec transactional(PlatformTransactionManager transactionManager) {
+		return transactional(transactionManager, false);
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} with the provided
+	 * {@code PlatformTransactionManager} and default {@link DefaultTransactionAttribute}
+	 * for the {@code pollingTask}.
+	 * @param transactionManager the {@link PlatformTransactionManager} to use.
+	 * @param handleMessageAdvice the flag to indicate the target {@link Advice} type:
+	 * {@code false} - regular {@link TransactionInterceptor}; {@code true} -
+	 * {@link org.springframework.integration.transaction.TransactionHandleMessageAdvice}
+	 * extension.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec transactional(PlatformTransactionManager transactionManager,
+			boolean handleMessageAdvice) {
```

Comment:
This method is redundant.
There is nothing like `MessageHandler` in this inbound adapter.
Therefore `TransactionHandleMessageAdvice` will work only fully the same way as regular `TransactionInterceptor`.

So, this variant is redundant and even confusing.

Or do you have something specific in mind since I don't see `transactional(boolean handleMessageAdvice)` variant.